### PR TITLE
🐛 fix: perms plugin - edit webhook

### PR DESCRIPTION
**Pull request by Yaikava**

### CODE REVIEW

File: helpers/webhooks/editWebhook.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import type { Webhook } from "../../types/webhooks/webhook.ts";
 import type { Bot } from "../../bot.ts";
 
 /** Edit a webhook. Requires the `MANAGE_WEBHOOKS` permission. Returns the updated webhook object on success. */
-export async function editWebhook(bot: Bot, webhookId: bigint, options: ModifyWebhook) {
+export async function editWebhook(bot: Bot, webhookId: bigint, options: ModifyWebhook, channelId?: bigint) {
```
Comment (lts20050703) [inline helpers/webhooks/editWebhook.ts:6]:
```suggestion
export async function editWebhook(bot: Bot, webhookId: bigint, options: ModifyWebhook) {
```

File: plugins/permissions/src/webhooks/editWebhook.ts [hunk 2]
```diff
@@ -4,8 +4,8 @@ import { requireBotChannelPermissions } from "../permissions.ts";
 export default function editWebhook(bot: BotWithCache) {
   const editWebhookOld = bot.helpers.editWebhook;
 
-  bot.helpers.editWebhook = async function (channelId, webhookId, options) {
-    requireBotChannelPermissions(bot, channelId, ["MANAGE_WEBHOOKS"]);
+  bot.helpers.editWebhook = async function (webhookId, options, channelId) {
```
Comment (lts20050703) [inline plugins/permissions/src/webhooks/editWebhook.ts:7]:
```suggestion
  bot.helpers.editWebhook = async function (webhookId, options) {
```

File: plugins/permissions/src/webhooks/editWebhook.ts [hunk 3]
```diff
@@ -4,8 +4,8 @@ import { requireBotChannelPermissions } from "../permissions.ts";
...
+    if (channelId) requireBotChannelPermissions(bot, channelId, ["MANAGE_WEBHOOKS"]);
```
Comment (lts20050703) [inline plugins/permissions/src/webhooks/editWebhook.ts:8]:
```suggestion
    if (options.channelId) requireBotChannelPermissions(bot, options.channelId, ["MANAGE_WEBHOOKS"]);
```

File: plugins/permissions/src/webhooks/editWebhook.ts [hunk 4]
```diff
@@ -18,6 +18,6 @@ export default function editWebhook(bot: BotWithCache) {
       }
     }
 
-    return await editWebhookOld(channelId, webhookId, options);
+    return await editWebhookOld(webhookId, options, channelId);
```
Comment (lts20050703) [inline plugins/permissions/src/webhooks/editWebhook.ts:21]:
```suggestion
    return await editWebhookOld(webhookId, options);
```

**Review by lts20050703 — CHANGES_REQUESTED** [review]
`ModifyWebhook` already has `channelId`
https://github.com/discordeno/discordeno/blob/7e69f29217e463a4bc6b0d8fb4336d360ce24c42/types/webhooks/modifyWebhook.ts#L8
image

**Comment by Skillz4Killz** [comment]
Overriden directly. Thanks yui, implemented your idea with a few modifications

**Pull request closed without merging**